src/libmusig2.c

```diff
@@ -1,68 +1,70 @@
                     break;
+                }
             }
         }
     }
     return 1;
 }
 
-static void musig2_key_agg_coef(musig2_context *mc, unsigned char *ser_pk, unsigned char *a) {
+static void musig2_key_agg_coef(musig2_context *mc, unsigned char *a, unsigned char *serialized_pubkey) {
 
     unsigned char tag[13] = "BIP0340/nonce";    // Tag of hash to generate the exponents
-    unsigned char temp_concat[(mc->nr_signers + 1) * MUSIG2_AGGR_PUBKEY_BYTES];    // Temp to store the concatenation of public keys
+    unsigned char temp_concat[(mc->nr_signers + 1) * MUSIG2_AGGR_PUBKEY_BYTES]; // Temp to store the concatenation of public keys
 
-    memcpy(temp_concat, mc->L, mc->nr_signers * MUSIG2_AGGR_PUBKEY_BYTES );      // Copy L into temp_concat
-    memcpy(&temp_concat[mc->nr_signers * MUSIG2_AGGR_PUBKEY_BYTES], ser_pk, MUSIG2_AGGR_PUBKEY_BYTES );  /* Copy given pk besides L */
+    memcpy(temp_concat, mc->L, mc->nr_signers * MUSIG2_AGGR_PUBKEY_BYTES ); /* Copy L into temp_concat */
+    memcpy(&temp_concat[mc->nr_signers * MUSIG2_AGGR_PUBKEY_BYTES], serialized_pubkey, MUSIG2_AGGR_PUBKEY_BYTES );  /* Copy given pk besides L */
     assert(secp256k1_tagged_sha256(mc->ctx, a, tag, sizeof (tag), temp_concat, sizeof (temp_concat)));
+
 }
 
-static void musig2_calc_b(musig2_context *mc, const unsigned char *ser_aggr_pk, unsigned char *b, const unsigned char *msg, int msg_len, int state) {
+static void musig2_calc_b(musig2_context *mc, unsigned char *b, const unsigned char *serialized_aggr_pubkey, const unsigned char *msg, int msg_len, int state) {
 
-    int j;
+    secp256k1_xonly_pubkey temp_xonly_pk;
     unsigned char tag[13] = "BIP0340/nonce";    // Tag for the hash to compute b
-    unsigned char ser_R[MUSIG2_AGGR_PUBKEY_BYTES];
-    unsigned char temp_concat[(1 + V) * MUSIG2_AGGR_PUBKEY_BYTES + msg_len]; // Temp value to store the concatenation of aggr_pk, aggr_R_list and the message.
-    secp256k1_xonly_pubkey xonly_R;
+    unsigned char serialized_aggr_R[MUSIG2_AGGR_PUBKEY_BYTES];
+    unsigned char temp_concat[(1 + V) * MUSIG2_AGGR_PUBKEY_BYTES + msg_len]; // Temp value to store the concatenation of aggr_pubkey, aggr_R_list and the message.
+    int j;
 
-    /* Copy ser_aggr_pk into temp_concat */
-    memcpy(temp_concat, ser_aggr_pk, MUSIG2_AGGR_PUBKEY_BYTES);
+    /* Copy serialized_aggr_pubkey into temp_concat */
+    memcpy(temp_concat, serialized_aggr_pubkey, MUSIG2_AGGR_PUBKEY_BYTES);
 
     /* Get x_only R_j, serialize and concatenate. */
     for (j = 0; j < V; j++) {
-        assert(secp256k1_xonly_pubkey_from_pubkey(mc->ctx, &xonly_R, NULL, mc->aggr_R_list[state * V + j]));
-        secp256k1_xonly_pubkey_serialize(mc->ctx, ser_R, &xonly_R);
-        memcpy(&temp_concat[MUSIG2_AGGR_PUBKEY_BYTES * (j + 1)], ser_R, MUSIG2_AGGR_PUBKEY_BYTES);
+        assert(secp256k1_xonly_pubkey_from_pubkey(mc->ctx, &temp_xonly_pk, NULL, mc->aggr_R_list[state * V + j]));
```

Comment:
Why did we rename this to `pk`, the comments says we are getting `R_j`.

Reply:
I thought that I should make generic variable names for the temp values. Since we are working with the `R_j` as `secp256k1_pubkey` I made this naming. But if you think it makes the code hard to understand I can change it. @iquerejeta.

Reply:
I think that if it is not a public key, the variable should not be named pubkey, even if its type is secp256k1_pubkey due to the underlying API.